Sources/Protocols/CellItem/CollectionViewCellItem.swift

```diff
@@ -0,0 +1,325 @@
+            ClosureWrapper<ActionResolver>.setHandler(newValue, for: self, key: &AssociatedKeys.shouldDeselectHandler)
+        }
+    }
+    
+    var itemWillDisplayCellHandler: CellActionHandler? {
+        get {
+            return ClosureWrapper<CellActionHandler>.handler(for: self, key: &AssociatedKeys.willDisplayCellHandler)
+        }
+        set {
+            ClosureWrapper<CellActionHandler>.setHandler(newValue, for: self, key: &AssociatedKeys.willDisplayCellHandler)
+        }
+    }
+    
+    var itemDidEndDisplayingCellHandler: CellActionHandler? {
+        get {
+            return ClosureWrapper<CellActionHandler>.handler(for: self, key: &AssociatedKeys.didEndDisplayingCellHandler)
+        }
+        set {
+            ClosureWrapper<CellActionHandler>.setHandler(newValue, for: self, key: &AssociatedKeys.didEndDisplayingCellHandler)
+        }
+    }
+    
+    var itemWillDisplayViewHandler: ViewActionHandler? {
+        get {
+            return ClosureWrapper<ViewActionHandler>.handler(for: self, key: &AssociatedKeys.willDisplayViewHandler)
+        }
+        set {
+            ClosureWrapper<ViewActionHandler>.setHandler(newValue, for: self, key: &AssociatedKeys.willDisplayViewHandler)
+        }
+    }
+    
+    var itemDidEndDisplayingViewHandler: ViewActionHandler? {
+        get {
+            return ClosureWrapper<ViewActionHandler>.handler(for: self, key: &AssociatedKeys.didEndDisplayingViewHandler)
+        }
+        set {
+            ClosureWrapper<ViewActionHandler>.setHandler(newValue, for: self, key: &AssociatedKeys.didEndDisplayingViewHandler)
+        }
+    }
+    
+    var itemCanMoveResolver: ActionResolver? {
+        get {
+            return ClosureWrapper<ActionResolver>.handler(for: self, key: &AssociatedKeys.canMoveHandler)
+        }
+        set {
+            ClosureWrapper<ActionResolver>.setHandler(newValue, for: self, key: &AssociatedKeys.canMoveHandler)
+        }
+    }
+    
+    // MARK: - Functions
+    
+    func shouldHighlight() -> Bool {
+        return itemShouldHighlightResolver?() ?? true
+    }
+    
+    func didHighlight() {
+        itemDidHighlightHandler?()
+    }
+    
+    func didUnhighlight() {
+        itemDidUnhighlightHandler?()
+    }
+    
+    func shouldSelect() -> Bool {
+        return itemShouldSelectResolver?() ?? true
+    }
+    
+    func shouldDeselect() -> Bool {
+        return itemShouldDeselectResolver?() ?? true
+    }
+    
+    func didSelect() {
+        itemDidSelectHandler?()
+    }
+    
+    func didDeselect() {
+        itemDidDeselectHandler?()
+    }
+    
+    func willDisplay(cell: UICollectionViewCell) {
+        itemWillDisplayCellHandler?(cell)
+    }
+    
+    func didEndDisplaying(cell: UICollectionViewCell) {
+        itemDidEndDisplayingCellHandler?(cell)
+    }
+    
+    func canMove() -> Bool {
+        return itemCanMoveResolver?() ?? false
+    }
+    
+    func size() -> CGSize {
+        return CGSize(width: 50, height: 50)
```

Comment:
Why 50x50? Maybe don't we need the default implementation?

Reply:
It is the default implementation 😄